Open API Endpoint for User Data

AKidd95 opened the pull request:
#### PT Story: https://www.pivotaltracker.com/story/show/156533623
#### Description
Changes proposed in this pull request:

* Open API enpoints to request user data from backend
* Test API requests successfully

#### What I have learned working on this feature: 
* how to use texture files
* how to do format request specs
* how to setup and configure devise_token_auth

OskarCA committed:
Initial local commit

OskarCA committed:
Initial local commit

OskarCA committed:
Merge branch 'develop' of https://github.com/CraftAcademy/languaship_backend into develop

AKidd95 committed:
generate User class
add token_auth namespace
gem devise_token_auth
create apicontroller

AKidd95 committed:
add request spec for user authentication
setup user api controller
add controller actions
Co-authored-by: Oskar https://github.com/OskarCA

AKidd95 committed:
add request spec for user

AKidd95 committed:
fix request specs

OskarCA committed:
x§Merge branch 'token_auth' of https://github.com/AKidd95/languaship_backend into token_auth
Merge fix

OskarCA committed:
Removed dotenv gem & remove .env from .gitignore untill needed

AKidd95 committed:
request users from api working

OskarCA committed:
Merge branch 'token_auth' of https://github.com/AKidd95/languaship_backend into token_auth

AKidd95 committed:
edit user factory

AKidd95 committed:
uncomment authenticate_user_api_v1!

OskarCA committed:
Merge branch 'token_auth' of https://github.com/AKidd95/languaship_backend into token_auth

AKidd95 committed:
add registrations spec for user

AKidd95 committed:
add user class to migration

AKidd95 committed:
change migration file

AKidd95 committed:
remove user class from migration

tochman commented on app/controllers/api/v1/user_controller.rb:
```diff
@@ -0,0 +1,18 @@
+
+  def users_near_you
+    @users = User.all
+    @user = User.find_by(id: params[:id])
```
I’m not sure I understand this

tochman commented on config/routes.rb:
```diff
@@ -1,7 +1,13 @@
     end
+
+    namespace :v1 do
+      resources :user, constraints: {format: 'json'}
```
Resources are usually set in plural.

AKidd95 replied:
Ok, this is good to know, When I tried to change it I got an `uninitialized constant API::V1:UserControlller` error. even after changing the name of the controller

tochman commented on spec/fixtures/files/user.txt:
```diff
@@ -0,0 +1,21 @@
+   "nickname" => "MeepMills",
+   "image" => nil,
+   "email" => "hello@itsme.com"
+  }}
```
I note that you are not using the JSON:API serialization standard. Why not?

tochman commented on spec/requests/api/v1/user_request_spec.rb:
```diff
@@ -0,0 +1,16 @@
+
+    it 'Should return all users' do
+      get '/api/v1/user', params: {
+        email: user.email, encrypted_password: user.encrypted_password
```
How can you send the encrypted password in params?  This don’t roller requires a user to be authenticated so you need to pass in credentials in headers.

tochman reviewed (commented):
I would like you to revisit this before merge.

OskarCA committed:
Merge fix

OskarCA committed:
Modiefied gemfile.lock

AKidd95 committed:
fix funky id in user controller
switch to JSON:API serializtion standard
pass email and password through headers

OskarCA committed:
Merge branch 'token_auth' of https://github.com/AKidd95/languaship_backend into token_auth

OskarCA committed:
Reafactoring of files & some cleaning up

AKidd95 committed:
format routes
add test for user id request

Fleurtam reviewed (approved):
looks good!

holgertidemand commented on spec/requests/api/v1/user_request_spec.rb:
```diff
@@ -0,0 +1,24 @@
+        email: user.email, password: user.password
+      }
+
+      expected_response = eval(file_fixture('user.txt').read)
```
You are actually not testing anything here. 
You need to add something like this after this line.
``` 
expect(object['data'].first).to eq expected_response
``` 
this just reads the user.txt file and sets expected_response equal to it :)
```
expected_response = eval(file_fixture('user.txt').read)
```

Just a heads up.

holgertidemand reviewed (commented):
Just a heads up :)
